ntoskrnl/cc/copy.c

```diff
@@ -381,6 +381,16 @@ CcCanIWrite (
         return FALSE;
     }
 
+    /* Otherwise, if there are no deferred writes yet, start the lazy writer */
+    if (IsListEmpty(&CcDeferredWrites))
+    {
+        KIRQL OldIrql;
+
+        OldIrql = KeAcquireQueuedSpinLock(LockQueueMasterLock);
+        CcScheduleLazyWriteScan(TRUE);
```

Comment:
The rules around this aren't clear to me. If you understand them, could you explain (and ideally, document in the code)?

* Do we need to check `LazyWriter.ScanActive`?
* What is the master lock protecting here?
* Don't we need to hold `CcDeferredWriteSpinLock` before checking `CcDeferredWrites`?
* Is this call broken since it _doesn't_ acquire the master lock? https://github.com/reactos/reactos/pull/4739/files#diff-e76ba4c26f0c04317701b9f1891af3634bfce59f91c80468c469eb18e58432a3L189